mono/mini/debugger-agent.c

```diff
@@ -10397,4 +10547,71 @@ mono_debugger_agent_parse_options (char *options)
 	sdb_options = options;
 }
 
+// Called from main thread
+static void burst_mono_shutdown()
+{
+	burst_lock();
+	// Leave the domain registered (as this prevents mono from trying to decode any mmethods belonging solely to burst)
+	// But we clear the callbacks, because the callbacks will be torn down in burst shortly
+	g_BurstDebugCallbacks.BurstFinishSetup = NULL;
+	g_BurstDebugCallbacks.BurstIsFileBursted = NULL;
+	g_BurstDebugCallbacks.BurstBreakpointNotify = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethods = NULL;
+	g_BurstDebugCallbacks.BurstFetchFiles = NULL;
+	g_BurstDebugCallbacks.BurstFetchDebugInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethodInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchParamInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethodName = NULL;
+	burst_unlock();
+}
+
+// Called from the main thread - once only
+static void burst_mono_install_hooks_imp(BurstMonoDebuggerCallbacks* callbacks,void* domainInitCallback)
+{
+	mono_coop_mutex_init_recursive(&g_BurstDebugMutex);
```

Comment:
Is it worth the special initialization here and checks for `g_BurstDebugMutexInitialised` in other places? I am wondering if we should just always initialize this mutex?

Reply:
Happy to just always init.